src/server/src/main/java/io/cassandrareaper/resources/ClusterResource.java

```diff
@@ -195,108 +185,102 @@ private Response addOrUpdateCluster(
       return Response.status(Response.Status.BAD_REQUEST).entity("query parameter \"seedHost\" required").build();
     }
 
-    try {
-      Cluster cluster = findClusterWithSeedHost(seedHost.get(), jmxPort);
-      if (null == cluster) {
-
-        String msg = String
-            .format("failed to find cluster %s with seed host %s", clusterName.orElse(""), seedHost.get());
-
-        return Response.status(Response.Status.BAD_REQUEST).entity(msg).build();
-      }
-      if (clusterName.isPresent() && !cluster.getName().equals(clusterName.get())) {
+    final Cluster cluster = findClusterWithSeedHost(seedHost.get(), jmxPort);
+    if (null == cluster) {
+      return Response
+          .status(Response.Status.BAD_REQUEST)
+          .entity(String.format("no cluster %s with seed host %s", clusterName.orElse(""), seedHost.get()))
+          .build();
+    }
+    if (clusterName.isPresent() && !cluster.getName().equals(clusterName.get())) {
+      String msg = String.format(
+          "POST/PUT on cluster resource %s called with seedHost %s belonging to different cluster %s",
+          clusterName.get(),
+          seedHost.get(),
+          cluster.getName());
+
+      LOG.info(msg);
+      return Response.status(Response.Status.BAD_REQUEST).entity(msg).build();
+    }
 
-        String msg = String.format(
-            "POST/PUT on cluster resource %s called with seedHost %s belonging to different cluster %s",
-            clusterName.get(),
-            seedHost.get(),
-            cluster.getName());
+    Supplier<Stream<Cluster>> matchingClusters
+        = () -> context.storage.getClusters().stream().filter(c -> c.getName().equalsIgnoreCase(cluster.getName()));
 
-        LOG.info(msg);
-        return Response.status(Response.Status.BAD_REQUEST).entity(msg).build();
-      }
-      Optional<Cluster> existingCluster = context.storage.getCluster(cluster.getName());
-      URI location = uriInfo.getBaseUriBuilder().path("cluster").path(cluster.getName()).build();
-      if (existingCluster.isPresent()) {
-        LOG.debug("Attempting updating nodelist for cluster {}", existingCluster.get().getName());
+    Preconditions.checkState(
+        1 >= matchingClusters.get().count(),
```

Comment:
I think this will prevent re-adding a cluster that's been put in `DELETED` state. There would be no way to add back a cluster that was mistakenly deleted.

Reply:
That's not what this precondition is checking, and it won't prevent re-adding an existing deleted cluster (so long as nodes overlap).